pkg/distributor/distributor.go

```diff
@@ -1289,15 +1289,8 @@ func (d *Distributor) push(ctx context.Context, pushReq *Request) error {
 	// Get a subring if tenant has shuffle shard size configured.
 	subRing := d.ingestersRing.ShuffleShard(userID, d.limits.IngestionTenantShardSize(userID))
 
-	// Use a background context to make sure all ingesters get samples even if we return early
-	localCtx, cancel := context.WithTimeout(context.Background(), d.cfg.RemoteTimeout)
-	localCtx = user.InjectOrgID(localCtx, userID)
-	// Get clientIP(s) from Context and add it to localCtx
-	source := util.GetSourceIPsFromOutgoingCtx(ctx)
-	localCtx = util.AddSourceIPsToOutgoingContext(localCtx, source)
-	if sp := opentracing.SpanFromContext(ctx); sp != nil {
-		localCtx = opentracing.ContextWithSpan(localCtx, sp)
-	}
```

Comment:
I verified that these operations are equivalent - using the value from `ctx` and injecting it anew is the same as using `ctx`.

cc @jhalterman since you asked